src/Orleans/Logging/LogWriters.cs

```diff
@@ -159,26 +159,29 @@ public class LogWriterToTrace : LogWriterBase, IFlushableLogConsumer
         /// <summary>Write the log message for this log.</summary>
         protected override void WriteLogMessage(string msg, Logger.Severity severity)
         {
+            var sev = SeverityLevel.Information;
+
             switch (severity)
             {
                 case Logger.Severity.Off:
-                    break;
+                    return;
                 case Logger.Severity.Error:
-                    Trace.TraceError(msg);
+                    sev = SeverityLevel.Error;
                     break;
                 case Logger.Severity.Warning:
-                    Trace.TraceWarning(msg);
+                    sev = SeverityLevel.Warning;
                     break;
                 case Logger.Severity.Info:
-                    Trace.TraceInformation(msg);
+                    sev = SeverityLevel.Information;
                     break;
                 case Logger.Severity.Verbose:
                 case Logger.Severity.Verbose2:
                 case Logger.Severity.Verbose3:
-                    Trace.WriteLine(msg);
+                    sev = SeverityLevel.Verbose;
                     break;
             }
-            Flush();
+            //Forward log calls to Telemetry API.
+            Logger.TelemetryConsumers.OfType<TraceTelemetryConsumer>().AsParallel().ForAll(tc => tc.TrackTrace(msg, sev));
```

Comment:
also, if you start switching threads, it may mean now the log statements will be out of order (don't think thread pool guarantees order and on the client we only have thread pool). Which is another reason not to switch threads and do all inline and if needed let the TelemetryConsumer do its own multi-threading.